### PR TITLE
Adding docs for debugger job support in operator

### PR DESCRIPTION
**Pull request by gautamkmr**
*Issue #, if available:*

*Description of changes:*
Adding docs for debugger support in operator

*Testing done:*
tox tests/unit has passed

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by gautamkmr** [comment]
@RedbackThomson @laurenyu

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 1]
```diff
@@ -304,6 +304,134 @@ job stops or completes.
 continue to show on the Amazon SageMaker console. The delete command
 takes about 2 minutes to clean up the resources from Amazon SageMaker.
 
+Sagemaker Debugger Jobs
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: nit: SageMaker

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 2]
```diff
@@ -304,6 +304,134 @@ job stops or completes.
...
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Create training job has an option to run asynchronous debugger job.
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: `When creating a SageMaker training job, you have an option to run asynchronous debugger jobs for your model`?

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 3]
```diff
@@ -304,6 +304,134 @@ job stops or completes.
...
+        Value:            tagValue
+      Training Job Name:  xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9
+    Status:
+      Cloud Watch Log URL:  https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logStream:group=/aws/sagemaker/TrainingJobs;prefix=xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9;streamFilter=typeLogStreamPrefix
+      Debug Rule Evaluation Statuses:
+        Last Modified Time:          2020-03-18T06:03:48Z
+        Rule Configuration Name:     LossNotDecreasing
+        Rule Evaluation Job Arn:     arn:aws:sagemaker:us-west-2:1234567890:processing-job/xgboost-mnist-debugger-8fe-lossnotdecreasing-a7d0eaf2
+        Rule Evaluation Status:      NoIssuesFound
+      Model Path:                    s3://sagemaker-us-west-2-1234567890/xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9/output/model.tar.gz
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: nit: Change bucket name to `my-bucket`

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 4]
```diff
@@ -304,6 +304,134 @@ job stops or completes.
...
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Create training job has an option to run asynchronous debugger job.
+It gives you full visibility into a training job by using a hook to capture
+tensors that define the state of the training process at each instance in
+its lifecycle. It also provides the capability of defining 'rules' to
+analyze the captured tensors. See \ `Sagemaker Debugger Introduction <https://docs.aws.amazon.com/sagemaker/latest/dg/train-debugger.html>`__ and \ `How Debugger Works <https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-how-it-works.html>`__ for details.
+
+You can get more details on debug job by using ``describe`` kubectl verb.
+Training job describe output will have ``Debug Rule Evaluation Statuses:``
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: `The output of describing a training job will now have a new field`

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 5]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
 continue to show on the Amazon SageMaker console. The delete command
 takes about 2 minutes to clean up the resources from Amazon SageMaker.
 
+SageMaker Debugger Jobs
+^^^^^^^^^^^^^^^^^^^^^^^
+
+When creating a SageMaker training job, you have an option to run 
+asynchronous debugger jobs for your model. It gives you full visibility 
+into a training job by using a hook to capturetensors that define
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: `capturetensors`

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 6]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
...
+        Volume Size In GB:  5
+      Role Arn:             arn:aws:iam::1234567890:role/service-role/AmazonSageMaker-ExecutionRole
+      Stopping Condition:
+        Max Runtime In Seconds:  86400
+      Tags:
+        Key:              tagKey
+        Value:            tagValue
+      Training Job Name:  xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9
+    Status:
+      Cloud Watch Log URL:  https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logStream:group=/aws/sagemaker/TrainingJobs;prefix=xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9;streamFilter=typeLogStreamPrefix
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: I don't think we need the full spec in this example. Perhaps just the Status field?

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 7]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
...
+        Rule Configuration Name:     LossNotDecreasing
+        Rule Evaluation Job Arn:     arn:aws:sagemaker:us-west-2:1234567890:processing-job/xgboost-mnist-debugger-8fe-lossnotdecreasing-a7d0eaf2
+        Rule Evaluation Status:      NoIssuesFound
+      Model Path:                    s3://my-bucket/sagemaker/xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9/output/model.tar.gz
+      Sage Maker Training Job Name:  xgboost-mnist-debugger-8fe3799e68dd11ea84231260529a8dc9
+      Secondary Status:              Completed
+      Training Job Status:           Completed
+    Events:                          <none>
+
+See \  `Sagemaker Debugger Examples <https://github.com/awslabs/amazon-sagemaker-examples/tree/master/sagemaker-debugger>`__ for more exampls of debugger jobs.
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: `Sagemaker` should be `SageMaker` and `exampls` should be `examples`

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 8]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
...
 
+SageMaker Debugger Jobs
+^^^^^^^^^^^^^^^^^^^^^^^
+
+When creating a SageMaker training job, you have an option to run 
+asynchronous debugger jobs for your model. It gives you full visibility 
+into a training job by using a hook to capturetensors that define 
+the state of the training process at each instance in its lifecycle. 
+It also provides the capability of defining 'rules' to
+analyze the captured tensors. See \ `Sagemaker Debugger Introduction <https://docs.aws.amazon.com/sagemaker/latest/dg/train-debugger.html>`__ and \ `How Debugger Works <https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-how-it-works.html>`__ for details.
```
Comment (RedbackThomson) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: `Sagemaker` should be `SageMaker`

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst (hunk 6 above)
Comment (gautamkmr) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: I will still want to have full description so that even `spec` is clear to customer.

**Comment by gautamkmr** [comment]
@laurenyu  can u please merge this ?

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 9]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
...
 
+SageMaker Debugger Jobs
+^^^^^^^^^^^^^^^^^^^^^^^
+
+When creating a SageMaker training job, you have an option to run 
+asynchronous debugger jobs for your model. It gives you full visibility 
+into a training job by using a hook to capture tensors that define 
+the state of the training process at each instance in its lifecycle. 
+It also provides the capability of defining 'rules' to
+analyze the captured tensors. See \ `SageMaker Debugger Introduction <https://docs.aws.amazon.com/sagemaker/latest/dg/train-debugger.html>`__ and \ `How Debugger Works <https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-how-it-works.html>`__ for details.
```
Comment (laurenyu) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: nit: you don't need the backslashes here

File: doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst [hunk 10]
```diff
@@ -304,6 +304,135 @@ job stops or completes.
...
+
+You can get more details on debug job by using ``describe`` kubectl verb.
```
Comment (laurenyu) [inline doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: * s/debug/a debugging (or do you mean a training job here?)
* s/using/using the

Comment (gautamkmr) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: It is a trainingjob which has `side job` (debug job).

Comment (laurenyu) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: gotcha. it might help to describe that since right now it reads "you can describe the debug job. here's a command that says training job" 😂

Comment (gautamkmr) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: Well, debug jobs lives inside the training job not separately so unfortunately I am stuck with the same command. :-/

Comment (laurenyu) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: I think you could just rephrase it to something like "You can get more details on a training job's debugging job..."

Comment (gautamkmr) [reply doc/amazon_sagemaker_operators_for_kubernetes_jobs.rst]: At the beginning of paragraph its clear that debug is being launched as part of training job so don't know if I should really make that change
`When creating a SageMaker training job, you have an option to run 
asynchronous debugger jobs for your model...`
